File: stack/2/2.js
var isValid = function(s) {
  let stack = [];
  const match = { "{": "}", "}": "{", "[": "]", "]": "[", "(": ")", ")": "(" };

  for (let ch of s) {
    if (stack.length == 0 || ch !== match[stack[stack.length - 1]]) {
      stack.push(ch);
    } else {
      stack.pop();
    }
  }
  return stack.length === 0;
};

File: stack/2/2.py
class Solution:
    def isValid(self, s: str) -> bool:
        match = {"{": "}", "}": "{", "[": "]", "]": "[", "(": ")", ")": "("}
        stack = []
        for i, ch in enumerate(s):
            if (not stack) or ch != match[stack[-1]]:
                stack.append(ch)
            else:
                stack.pop()
        return not stack
